service/ntfy/ntfy.go

```diff
@@ -0,0 +1,174 @@
+package ntfy
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// Service allow you to configure Ntfy service.
+type Service struct {
+	client     *http.Client
+	serverURLs []string
+}
+
+func defaultHTTPClient() *http.Client {
+	return &http.Client{
+		Timeout: 5 * time.Second,
+	}
+}
+
+// DefaultServerURL is the default server to use for the Ntfy service.
+const DefaultServerURL = "https://ntfy.sh"
+
+// normalizeServerURL normalizes the server URL. It prefixes it with https:// if it's not already and appends a slash
+// if it's not already there. If the serverURL is empty, the DefaultServerURL is used. We're not validating the url here
+// on purpose, we leave that to the http client.
+func normalizeServerURL(serverURL string) string {
+	if serverURL == "" {
+		return DefaultServerURL
+	}
+
+	// Normalize the url
+	if !strings.HasPrefix(serverURL, "http") {
+		serverURL = "https://" + serverURL
+	}
+	if !strings.HasSuffix(serverURL, "/") {
+		serverURL = serverURL + "/"
+	}
+
+	return serverURL
+}
+
+// AddReceivers adds server URLs to the list of servers to use for sending messages.
+func (s *Service) AddReceivers(serverURLs ...string) {
+	for _, serverURL := range serverURLs {
+		serverURL = normalizeServerURL(serverURL)
+		s.serverURLs = append(s.serverURLs, serverURL)
+	}
+}
+
+// NewWithServers returns a new instance of Ntfy service. You can use this service to send messages to Ntfy. You can
+// specify the servers to send the messages to. By default, the service will use the default server
+// (https://api.day.app/) if you don't specify any servers.
+func NewWithServers(serverURLs ...string) *Service {
```

Comment:
@svaloumas @pushkar803 I'm torn about this one. On one hand I like the convenience this offers (we're doing the same in the [notify package](https://github.com/nikoksr/notify/blob/1e6d6681793098a26455771a55f200921477cad3/notify.go#L71)), but on the other hand I see two things I dislike.

1. No other service (to my knowledge) implements this type of function so far, which makes the library less predictable and intuitiv. So either, we adapt this pattern and extend all or most other services with the same function or we remove it from here.
2. The inconsistent naming. We have a `AddReceivers` and `NewWithServers`, the latter calls `AddReceivers` internally. I think, in case we stick with it, this should be renamed to `NewWithReceivers` to stay consistent and intuitiv.
3. A call to `NewWithServers` without is functionally identical to just calling `New` with an empty parameter list. So maybe, we can resolve my issue above (no. 2) by dropping `NewWithServers` entirely and extend `New` with a variadic receivers list. However, this again makes the package not follow our current behavioral standards, which makes it less predictable. Maybe it's better if we stay explicit with two seperate constructor functions as it currently is the case. Feedback on this please!